src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs

```diff
@@ -243,11 +243,20 @@ public RestoreCommand(FrameworkName fallbackFramework)
             var projectLockFilePath = Path.Combine(projectFolder, LockFileFormat.LockFileName);
 
             Runtime.Project project;
-            if (!Runtime.Project.TryGetProject(projectJsonPath, out project))
+            var diagnostics = new List<ICompilationMessage>();
+            if (!Runtime.Project.TryGetProject(projectJsonPath, out project, diagnostics))
             {
                 throw new Exception("TODO: project.json parse error");
             }
 
+            if (diagnostics.HasErrors())
+            {
+                var exceptions = diagnostics
+                    .Where(x => x.Severity == CompilationMessageSeverity.Error)
+                    .Select(x => new Exception(x.FormattedMessage));
+                throw new AggregateException(exceptions);
```

Comment:
I cannot see any benefit postponing the failure to build/publish. so let's fail fast to eliminate unnecessary confusion. Just pushed the fix.